Library/Homebrew/utils/ruby.sh

```diff
@@ -45,15 +56,29 @@ EOS
       then
         HOMEBREW_RUBY_PATH="/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby"
       else
-        HOMEBREW_RUBY_PATH="$(type -P ruby)"
+        HOMEBREW_RUBY_PATH=$(type -P ruby)
+        if [[ $(test-ruby "$HOMEBREW_RUBY_PATH") != "true" ]]
+        then
+          HOMEBREW_RUBY_PATH=$(PATH="$HOMEBREW_PATH" type -P ruby)
+          if [[ $(test-ruby "$HOMEBREW_RUBY_PATH") != "true" ]]
+          then
+            if [[ $old_ruby_usable != true ]]
```

Comment:
Would be good to avoid quite so much nesting here and instead use an intermediate variable(s) and multiple `if`s at the same level?

Reply:
Converted nested `if`s to a single if/elif/else conditional block in b4267d80e.
Had to add a line to `test-ruby` that checks that tested file exists and is executable.